rust/operator-binary/src/controller.rs

```diff
@@ -0,0 +1,890 @@
+                .apply_patch(
+                    FIELD_MANAGER_SCOPE,
+                    &rg_catalog_configmap,
+                    &rg_catalog_configmap,
+                )
+                .await
+                .with_context(|| ApplyRoleGroupConfig {
+                    rolegroup: rolegroup.clone(),
+                })?;
+
+            client
+                .apply_patch(FIELD_MANAGER_SCOPE, &rg_stateful_set, &rg_stateful_set)
+                .await
+                .with_context(|| ApplyRoleGroupStatefulSet {
+                    rolegroup: rolegroup.clone(),
+                })?;
+        }
+    }
+
+    Ok(ReconcilerAction {
+        requeue_after: None,
+    })
+}
+
+/// The coordinator-role service is the primary endpoint that should be used by clients that do not
+/// perform internal load balancing, including targets outside of the cluster.
+pub fn build_coordinator_role_service(trino: &TrinoCluster) -> Result<Service> {
+    let role_name = TrinoRole::Coordinator.to_string();
+    let role_svc_name = trino
+        .coordinator_role_service_name()
+        .context(GlobalServiceNameNotFound)?;
+    Ok(Service {
+        metadata: ObjectMetaBuilder::new()
+            .name_and_namespace(trino)
+            .name(&role_svc_name)
+            .ownerreference_from_resource(trino, None, Some(true))
+            .context(ObjectMissingMetadataForOwnerRef)?
+            .with_recommended_labels(trino, APP_NAME, trino_version(trino)?, &role_name, "global")
+            .build(),
+        spec: Some(ServiceSpec {
+            ports: Some(service_ports()),
+            selector: Some(role_selector_labels(trino, APP_NAME, &role_name)),
+            type_: Some("NodePort".to_string()),
+            ..ServiceSpec::default()
+        }),
+        status: None,
+    })
+}
+
+/// The rolegroup [`ConfigMap`] configures the rolegroup based on the configuration given by the administrator
+fn build_rolegroup_config_map(
+    trino: &TrinoCluster,
+    _role: &TrinoRole,
+    rolegroup_ref: &RoleGroupRef<TrinoCluster>,
+    config: &HashMap<PropertyNameKind, BTreeMap<String, String>>,
+    opa_connect: &Option<String>,
+) -> Result<ConfigMap> {
+    let mut cm_conf_data = BTreeMap::new();
+
+    // TODO: create via product config?
+    let mut jvm_config = "-server
+        -Xmx16G
+        -XX:-UseBiasedLocking
+        -XX:+UseG1GC
+        -XX:G1HeapRegionSize=32M
+        -XX:+ExplicitGCInvokesConcurrent
+        -XX:+ExitOnOutOfMemoryError
+        -XX:+HeapDumpOnOutOfMemoryError
+        -XX:-OmitStackTraceInFastThrow
+        -XX:ReservedCodeCacheSize=512M
+        -XX:PerMethodRecompilationCutoff=10000
+        -XX:PerBytecodeRecompilationCutoff=10000
+        -Djdk.attach.allowAttachSelf=true
+        -Djdk.nio.maxCachedBufferSize=2000000"
+        .to_string();
+
+    // TODO: we support only one coordinator for now
+    // TODO: remove unwrap
+    let coordinator_ref: TrinoPodRef = trino.coordinator_pods().unwrap().next().unwrap();
```

Comment:
Probably want to promote this to an Error at least, so we don't crash the whole process for invalid TrinoClusters.

Reply:
done